samples/subsys/video/capture_to_lvgl/src/main.c

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright (c) 2024 Charles Dias <charlesdias.cd@outlook.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/display.h>
+#include <zephyr/drivers/video.h>
+#include <lvgl.h>
+
+#define LOG_LEVEL CONFIG_LOG_DEFAULT_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(main);
+
+#define VIDEO_DEV_SW "VIDEO_SW_GENERATOR"
+
+void draw_label(void)
+{
+	lv_obj_t *label = lv_label_create(lv_scr_act());
+
+	lv_label_set_text(label, "LVGL on Zephyr!");
+	lv_obj_align(label, LV_ALIGN_CENTER, 0, 30);
+
+	lv_task_handler();
+}
+
+void draw_lvgl_logo(void)
+{
+	LV_IMG_DECLARE(img_lvgl_logo);
+	lv_obj_t *img1 = lv_img_create(lv_scr_act());
+
+	lv_img_set_src(img1, &img_lvgl_logo);
+	lv_obj_align(img1, LV_ALIGN_CENTER, 0, -10);
+
+	lv_task_handler();
+}
+
+void display_splash_screen(void)
+{
+	draw_label();
+	draw_lvgl_logo();
+
+	k_sleep(K_MSEC(5000));
+	lv_obj_clean(lv_scr_act());
+}
+
+int main(void)
+{
+	struct video_buffer *buffers[2], *vbuf;
+	const struct device *display_dev;
+	struct video_format fmt;
+	struct video_caps caps;
+	const struct device *video_dev;
+	unsigned int frame = 0;
+	size_t bsize;
+	int i = 0;
+
+	display_dev = DEVICE_DT_GET(DT_CHOSEN(zephyr_display));
+	if (!device_is_ready(display_dev)) {
+		LOG_ERR("Device not ready, aborting test");
+		return 0;
+	}
+
+#if DT_HAS_CHOSEN(zephyr_camera)
+	video_dev = DEVICE_DT_GET(DT_CHOSEN(zephyr_camera));
+	if (!device_is_ready(video_dev)) {
+		LOG_ERR("%s device is not ready", video_dev->name);
+		return 0;
+	}
+#else
+	video_dev = device_get_binding(VIDEO_DEV_SW);
+	if (video_dev == NULL) {
+		LOG_ERR("%s device not found", VIDEO_DEV_SW);
+		return 0;
+	}
+#endif
+
+	printk("- Device name: %s\n", video_dev->name);
+
+	/* Get capabilities */
+	if (video_get_caps(video_dev, VIDEO_EP_OUT, &caps)) {
+		LOG_ERR("Unable to retrieve video capabilities");
+		return 0;
+	}
+
+	printk("- Capabilities:\n");
+	while (caps.format_caps[i].pixelformat) {
+		const struct video_format_cap *fcap = &caps.format_caps[i];
+		/* fourcc to string */
+		printk("  %c%c%c%c width [%u; %u; %u] height [%u; %u; %u]\n",
```

Comment:
please log with appropriate level instead of using printk

Reply:
Hi @kartben, thanks for your review!

Should I declare the LOG_MODULE_REGISTER macro to this code?

Reply:
Changed to LOG_INF